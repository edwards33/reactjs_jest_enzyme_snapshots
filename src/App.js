import React from 'react';
import logo from './logo.svg';
import './App.css';

const Header = ({text}) => <div>{text}</div>

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <ul>
          <li>1. item</li>
          <li>2. item</li>
        </ul>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
      <Header text="New Header" />
    </div>
  );
}

export default App;
