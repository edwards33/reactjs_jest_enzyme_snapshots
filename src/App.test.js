import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({
  adapter: new Adapter()
});

describe('<App />', () => {
  const wrapper = shallow(<App />);
  it('a contains Learn React',() => {
    expect(wrapper.find('a').text()).toBe('Learn React');
  });
  it('equals to the snapshot',() => {
    expect(wrapper).toMatchSnapshot();
  });
  it('equals to the snapshot as Json',() => {
    const wrapperForSnapshot = shallow(
      <strong>Hello World!</strong>
    );
    expect(toJson(wrapperForSnapshot)).toMatchSnapshot();
  });
});
