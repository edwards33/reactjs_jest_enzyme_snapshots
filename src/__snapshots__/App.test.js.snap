// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> equals to the snapshot 1`] = `ShallowWrapper {}`;

exports[`<App /> equals to the snapshot as Json 1`] = `
<strong>
  Hello World!
</strong>
`;
